### PR TITLE
Message Card

### LINKED ISSUE #19: Message Card
image

### PR DESCRIPTION
**Pull request by LiAustin059**
Added a message card function that takes title, message, and author and outputs it like this.
Not finalized  some issues there, just a base of what I have that works atm
image

closes #19

### CODE REVIEW

**Comment by RedMaps** [comment]
Just saying, you might wanna merge main into this soon, looks like its way behind rn ^^

**Comment by LiAustin059** [comment]
Will do thanks for the feedback, ignore last three commits trying to pull from main, trying to figure this out.

**Comment by LiAustin059** [comment]
Will push from my main branch after this commit, just figuring things out.

**Comment by LiAustin059** [comment]
Here it looks like so far:
image
Ignore the deleting of variable file as I thought it would delete it from commit not in total.

**Comment by RedMaps** [comment]
This seems to have multiple build errors right now. I get this when i start dev: 
image
Also just from the pictures the spacing of the text seems to be off quite a bit as well as the thing at the top right being a bit too big.

**Comment by LiAustin059** [comment]
Sorry about that, thanks for the feedback.  
image
Here is how it looks like currently on desktop.

**Comment by nyxordinal** [comment]
image
Is it supposed to be like this? or the two columns should be in the center of the screen?

**Comment by nyxordinal** [comment]
Also when I try to build the frontend an error still pops up
image

**Comment by LiAustin059** [comment]
Fixed the current issues however found out that it doesn't look as wanted with empty space on text and on mobile doesn't look proper as well so working on that

**Comment by LiAustin059** [comment]
Finished the message card page, hopefully last commit. 
image
Something to note is that react stack grid needs to be install in order to run, not sure if something does it automatically once the package.json file is changed.

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "lerna": "^4.0.0",
     "prettier": "2.3.1",
     "typescript": "^4.3.2"
```
Comment (nyxordinal) [inline package.json]: ``
```suggestion
    "typescript": "^4.3.2",
    "react-stack-grid": "^0.7.1"
```

Comment (nyxordinal) [reply package.json]: minor typo in package.json

**Comment by nyxordinal** [comment]
image
Sorry, this is what it looks like when I try to run it on my local machine, am I missing somewhere?

**Comment by azuraga** [comment]
I have the same inconsistency with card spacing.

**Comment by azuraga** [comment]
> I have the same inconsistency with card spacing. 

Update on the spacing: it seems to happen upon first fresh load of the page, and gets resolved upon reload and resizing the browser window.

File: packages/frontend/components/MessageCards/MessageCard.module.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  img {
+    float: right;
+    position: relative;
+    top: -60px;
+    left: 10px;
+    height: 105px;
+  }
+
+  .message {
+    font-size: 24;
```
Comment (azuraga) [inline packages/frontend/components/MessageCards/MessageCard.module.scss]: Missing unit. I think you mean `px`, but that seems to be a bit too big compared to what you had before. `16px` works fine.

**Review by azuraga — CHANGES_REQUESTED** [review]
I suggested a thing about the `font-size` unit. Also, the fonts don't have a `serif`, `sans-serif`, etc. attached, which, together with the `font-size` thing, might have caused some undefined behavior.

Also, isn't it better to just have an array of card data, instead of an array of *JSONs* containing card data?

**Comment by azuraga** [comment]
Since the spacing issue has to do with animation, the optimized production build might automatically resolve it.

**Comment by LiAustin059** [comment]
> I suggested a thing about the `font-size` unit. Also, the fonts don't have a `serif`, `sans-serif`, etc. attached, which, together with the `font-size` thing, might have caused some undefined behavior.
> 
> Also, isn't it better to just have an array of card data, instead of an array of _JSONs_ containing card data?

Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.

File: package.json [hunk 3]
```diff
@@ -37,7 +37,9 @@
     "husky": "^6.0.0",
     "lerna": "^4.0.0",
     "prettier": "2.3.1",
-    "typescript": "^4.3.2"
+    "typescript": "^4.3.2",
+    "react-stack-grid": "^0.7.1"
+    "react-stack-grid": "^0.7.1",
```
Comment (azuraga) [inline package.json]:
```suggestion

```

**Review by azuraga — COMMENTED** [review]
> Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.

How's progress on this?

**Comment by LiAustin059** [comment]
> > Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.
> 
> How's progress on this?

Just finished it, sorry about being late was busy all day. Free in the next couple of days can work more whenever needed.

File: packages/frontend/pages/messages/index.tsx [hunk 4]
```diff
@@ -1,11 +1,55 @@
...
+    author: '-Author',
+  },
+]
 
-export default function Messages() {
+export default function MessageBoard() {
   return (
     <>
       <Navigation title="Messages" page={Page.MESSAGES} />
+      <MessageList key={'1'} cardData={Messages} />
```
Comment (azuraga) [inline packages/frontend/pages/messages/index.tsx]: Key isn't necessary in this case.
```suggestion
      <MessageList cardData={Messages} />
```

**Review by azuraga — CHANGES_REQUESTED** [review]
> sorry about being late was busy all day

No worries! I have one small change request, and then I think it's ready to be merged.

**Comment by nyxordinal** [comment]
Hi! It seems that the gap bug still exists when the page is first loaded, maybe you want to consider using `masonry-board` as in #73 ?

**Comment by LiAustin059** [comment]
Will do, I'll get working on that

**Comment by LiAustin059** [comment]
Messages page works now on fresh load and looks like this 
image

**Comment by azuraga** [comment]
Doesn't work properly on mobile. I'm looking for a fix
image

**Comment by azuraga** [comment]
> Doesn't work properly on mobile. I'm looking for a fix
> image

update: fix lays in modifying `styles.container`

**Comment by azuraga** [comment]
It might be better to inherit the Fanart MasonryBoard instead of creating a new one that barely has any difference. With that we can also easily fix bugs like this.

**Comment by LiAustin059** [comment]
Will do, at first thought it was better to make it two separate things to make it easier to make small changes.

**Comment by azuraga** [comment]
> Will do, at first thought it was better to make it two separate things to make it easier to make small changes.

Indeed, but since the two are pretty much the same concept, we could add those modifications in a different manner. I've incorporated using the fanart board for messages, too, will suggest the code in a minute.

File: packages/frontend/components/MessageCards/MessageList.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+export default function MessageList(props: MessageProps) {
+  const cards = props.cardData.map((message, i) => (
+    <MessageCard key={i} {...message} />
+  ))
+  return (
+    <div className={styles.container}>
+      <MasonryBoard> {cards}</MasonryBoard>
+    </div>
+  )
+}
```
Comment (azuraga) [inline packages/frontend/components/MessageCards/MessageList.tsx]:
```suggestion
import React from 'react'
import MessageCard from './MessageCard'
import MessageProps from '../../interfaces/messageProps'
import styles from '../Fanart/styles.module.scss'
import MasonryBoard from '../MasonryBoard'

export default function MessageList(props: MessageProps) {
  const cards = props.cardData.map((message, i) => (
    <MessageCard key={i} {...message} />
  ))
  return (
    <div className={styles.fanart_board}>
      <MasonryBoard> {cards}</MasonryBoard>
    </div>
  )
}
```

**Review by azuraga — CHANGES_REQUESTED** [review]
Fix for the message card misalignment. Will use 3 messages per row, can be adjusted once `MasonryBoard` is properly implemented. This will be done in a future PR.

**Comment by azuraga** [comment]
Alright one small thing. The one difference between fanart and messages in that board seems to be the breakpoints, so can you add a `breakpointCols` parameter to `MasonryBoard` and have `FanartBoard` and `MessageBoard` use appropriate parameters? For `FanartBoard` you can copy the `breakpointColsObj` from `MasonryBoard`, for `MessageBoard` the following seems to work:

```js
const breakpointColumnsObj = {
  default: 3,
  1160: 2,
  860: 1,
}
```

Also, the unused MasonryBoard stuff in the MessageCard can be removed.

**Comment by LiAustin059** [comment]
Sure shouldn't take too long

**Review by azuraga — APPROVED** [review]

**Comment by nyxordinal** [comment]
Please try to merge the latest commit from main branch again into this PR

**Pull request merged**